exchange/exchange.go

```diff
@@ -788,25 +788,10 @@ func (e *exchange) makeExtBidResponse(adapterBids map[openrtb_ext.BidderName]*pb
 // Return an openrtb seatBid for a bidder
 // BuildBidResponse is responsible for ensuring nil bid seatbids are not included
 func (e *exchange) makeSeatBid(adapterBid *pbsOrtbSeatBid, adapter openrtb_ext.BidderName, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, auc *auction, returnCreative bool) *openrtb.SeatBid {
-	seatBid := new(openrtb.SeatBid)
-	seatBid.Seat = adapter.String()
-	// Prebid cannot support roadblocking
-	seatBid.Group = 0
-
-	if len(adapterBid.ext) > 0 {
-		sbExt := ExtSeatBid{
-			Bidder: adapterBid.ext,
-		}
-
-		ext, err := json.Marshal(sbExt)
-		if err != nil {
-			extError := openrtb_ext.ExtBidderError{
-				Code:    errortypes.ReadCode(err),
-				Message: fmt.Sprintf("Error writing SeatBid.Ext: %s", err.Error()),
-			}
-			adapterExtra[adapter].Errors = append(adapterExtra[adapter].Errors, extError)
-		}
-		seatBid.Ext = ext
+	seatBid := &openrtb.SeatBid{
+		Seat:  adapter.String(),
+		Group: 0, // Prebid cannot support roadblocking
+		Ext:   adapterBid.ext,
```

Comment:
We were also doing this at the seatbid level, for reasons. It's completely unused currently and therefore safe to change.

Reply:
I realize this is beyond the scope of this PR but, I tried forcing `adapterBid.ext` to be non-nil by adding an `ext` to the mock response in one of your json tests:
```
45             "mockResponse": {
46                 "pbsSeatBid": {
   +                 "seat": "seatName",
   +                 "currency": 0,
   +                 "ext": {"someOtherField": "someOtherValue"},
47                     "pbsBids": [{
48                         "ortbBid": {
49                             "id": "apn-bid",
50                             "impid": "my-imp-id",
51                             "price": 0.3,
exchange/exchangetest/bid-ext-prebid-collision.json
```
But still this `adapterBid.ext` was `nil`.
```
╰─➤  dlv test
Type 'help' for list of commands.
(dlv) break exchange/exchange_test.go:1535
Breakpoint 1 set at 0x1a13b43 for github.com/prebid/prebid-server/exchange.TestExchangeJSON() ./exchange_test.go:1535
(dlv) cond 1 fileDisplayName == "exchange/exchangetest/bid-ext-prebid-collision.json"
(dlv) c
> github.com/prebid/prebid-server/exchange.TestExchangeJSON() ./exchange_test.go:1535 (hits goroutine(1036):1 total:1) (PC: 0x1a13b43)
  1530: func TestExchangeJSON(t *testing.T) {
  1531:         if specFiles, err := ioutil.ReadDir("./exchangetest"); err == nil {
  1532:                 for _, specFile := range specFiles {
  1533:                         fileName := "./exchangetest/" + specFile.Name()
  1534:                         fileDisplayName := "exchange/exchangetest/" + specFile.Name()
=>1535:                         t.Run(fileDisplayName, func(t *testing.T) {
  1536:                                 specData, err := loadFile(fileName)
  1537:                                 if assert.NoError(t, err, "Failed to load contents of file %s: %v", fileDisplayName, err) {
  1538:                                         runSpec(t, fileDisplayName, specData)
  1539:                                 }
  1540:                         })
(dlv) p fileDisplayName
"exchange/exchangetest/bid-ext-prebid-collision.json"
(dlv) break exchange/exchange.go:794
Breakpoint 2 set at 0x19c05e3 for github.com/prebid/prebid-server/exchange.(*exchange).makeSeatBid() ./exchange.go:794
(dlv) c
> github.com/prebid/prebid-server/exchange.(*exchange).makeSeatBid() ./exchange.go:794 (hits goroutine(1040):1 total:1) (PC: 0x19c05e3)
   789: // BuildBidResponse is responsible for ensuring nil bid seatbids are not included
   790: func (e *exchange) makeSeatBid(adapterBid *pbsOrtbSeatBid, adapter openrtb_ext.BidderName, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, auc *auction, returnCreative bool) *openrtb.SeatBid {
   791:         seatBid := &openrtb.SeatBid{
   792:                 Seat:  adapter.String(),
   793:                 Group: 0, // Prebid cannot support roadblocking
=> 794:                 Ext:   adapterBid.ext,
   795:         }
   796:
   797:         var errList []error
   798:         seatBid.Bid, errList = e.makeBid(adapterBid.bids, auc, returnCreative)
   799:         if len(errList) > 0 {
(dlv) p adapterBid.ext
encoding/json.RawMessage len: 0, cap: 0, nil
```
For this and every other test:
```
╰─➤  dlv test
Type 'help' for list of commands.
(dlv) break exchange/exchange.go:794
Breakpoint 1 set at 0x19c05e3 for github.com/prebid/prebid-server/exchange.(*exchange).makeSeatBid() ./exchange.go:794
(dlv) cond 1 adapterBid.ext != nil
(dlv) c
PASS
Process 52260 has exited with status 0
(dlv) q
```
Should we add a coded test case where `adapterBid.ext` comes with a value? Or, If we don't plan to use this field at all, should we just remove the assignment in line 794 so it is clear it'll always be `nil`?
```
790   func (e *exchange) makeSeatBid(adapterBid *pbsOrtbSeatBid, adapter openrtb_ext.BidderName, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, auc *auction, returnCreative bool) *openrtb.SeatBid {
791           seatBid := &openrtb.SeatBid{
792                   Seat:  adapter.String(),
793                   Group: 0, // Prebid cannot support roadblocking
794  -                Ext:   adapterBid.ext,
795           }
796  
797           var errList []error
798           seatBid.Bid, errList = e.makeBid(adapterBid.bids, auc, returnCreative)
799           if len(errList) > 0 {
exchange/exchange.go
```

Reply:
Good point. There doesn't seem to be any way for an adapter to set the seatbid.ext. Removed.